internal/conf/configuration.go

```diff
@@ -648,53 +648,71 @@ func LoadGlobal(filename string) (*GlobalConfiguration, error) {
 	}
 
 	config := new(GlobalConfiguration)
+	if err := loadGlobal(config); err != nil {
+		return nil, err
+	}
+	return config, nil
+}
 
+func LoadGlobalFiles(filenames ...string) (*GlobalConfiguration, error) {
```

Comment:
Just double making sure, this will load config like:


1. Environment
2. File A
3. File B
4. File C

Right? Environment goes first?

Reply:
Correct, but the Environment is only used for startup. It is overwritten since godotenv writes to Go's copy of the process environment with os.Setenv. I didn't touch any of the logic for starting the process. I was thinking the purpose of keeping the -c flag was for the initialization and the config dir was for runtime changes that overwrote it. But maybe it shouldn't work this way?

How it works now:
```sh
# process
export GOTRUE_base_env_var_01=...
export GOTRUE_base_env_var_01=...

auth -c '/base.conf' -w '/auth.d'
```

Before the first request:
 - os.Environ() already contains process values
	- `conf.LoadGlobal("/base.conf")`
	- /base.conf is read, overriding any env vars (by godotenv.Override)
 - OR
	- `conf.LoadGlobal("")` (-c not flag empty or missing)
	- .env is optionally loading, env vars have precedence (godotenv.Load)

The root handler is constructed and begins serving requests with this initialization configuration. Nothing is touched in the `-watch-dir` directory. If later on someone adds files there, they will start taking precedence. Since godotenv writes to the environment, they are lost if overwritten.

If auth.d should be part of initialization above we should probably make the two flags mutually exclusive, or define the ordering behavior concretely. That probably makes more sense come to think of it.

On both initialization and config.Reload, do the same exact behavior:
- conf.LoadGlobalFiles(base, configDirFiles...)
  - first if base is "" (do as we do now)
    - check for ".env", don't override env vars
    - load base, override all env vars
  - second if configDirFiles is not empty, load each one in sequence